### PR TITLE
feat: shared cache for next caching

### PR DESCRIPTION
**Pull request by ShubhamPalriwala**
Closes https://github.com/formbricks/internal/issues/54

## What does this PR do?
Uses a combination of:
- Redis (for next caching)
- Webdis (HTTP wrapper over redis) (for rate limiting in middleware due to the edge runtime)

A new env var:
```
REDIS_HTTP_INTERFACE_URL
```

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues
- [ ] First PR at Formbricks? [Please sign the CLA!](https://cla-assistant.io/formbricks/formbricks) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: apps/web/cache-handler.mjs [hunk 1]
```diff
@@ -5,11 +5,13 @@ import { createClient } from "redis";
...
 CacheHandler.onCreation(async () => {
   let redisHandler;
-  if (process.env.REDIS_CLIENT_URL) {
+  if (process.env.REDIS_URL) {
     const client = createClient({
-      url: process.env.REDIS_CLIENT_URL,
+      url: process.env.REDIS_URL,
+    });
+    client.on("error", (e) => {
+      console.log("Error in conncting to Redis client", e);
```
Comment (mattinannt) [inline apps/web/cache-handler.mjs:13]: please change to console.error

File: kamal/deploy.yml [hunk 2]
```diff
@@ -170,6 +171,21 @@ accessories:
     port: "172.31.40.79:6379:6379"
     directories:
       - data:/data
+
+  webdis:
+    image: nicolas/webdis:latest
```
Comment (mattinannt) [inline kamal/deploy.yml:176]: please pin version number

**Review by mattinannt — CHANGES_REQUESTED** [review]
@ShubhamPalriwala thanks :-) looks much better and simpler than before!
A few small points:

File: apps/web/cache-handler.mjs [hunk 3]
```diff
@@ -5,11 +5,13 @@ import { createClient } from "redis";
 
 CacheHandler.onCreation(async () => {
   let redisHandler;
-  if (process.env.REDIS_CLIENT_URL) {
+  if (process.env.REDIS_URL) {
```
[the remark this answers is not in the document]
Comment (ShubhamPalriwala) [reply apps/web/cache-handler.mjs:8]: does not allow in next config direct imported file

File: kamal/deploy.yml (hunk 2 above)
Comment (ShubhamPalriwala) [reply kamal/deploy.yml]: done

**Review by mattinannt — APPROVED** [review]
@ShubhamPalriwala Looks great! Thanks for making the changes :-)

**Pull request merged**